src/middlewared/middlewared/plugins/dyndns.py

```diff
@@ -15,6 +15,46 @@ class Config:
         dyndns["domain"] = dyndns["domain"].replace(',', ' ').replace(';', ' ').split()
         return dyndns
 
+    @accepts()
+    async def list_providers(self):
```

Comment:
Please follow our not written standard across all other plugins ; )

`provider_choices`